### ISSUE TITLE
Add general and source code attributes to spans

### ISSUE THREAD

**Issue by tsloughter** [open-telemetry/opentelemetry-erlang#517]
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#source-code-attributes

**Comment by bullno1** [comment]
I kinda need this too.

This is only possible in macro right?

**Comment by tsloughter** [comment]
Correct, it would need to be done in the macros for `with_span` and `start_span`.

**Comment by bullno1** [comment]
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-thread-attributes

On the topic of `thread.id` attribute, `erlang:system_info(scheduler_id)` is suggested.
While this is kind of correct, I'm not sure how useful it is compared to say: `self()`.
But formatting `self()` can have some overhead that is not always desirable.

**Comment by bullno1** [comment]
On the same topic `code.function` also requires some string/binary construction from `?FUNCTION_NAME` and `?FUNCTION_ARITY`.
It's probably negligible but it's not nothing, even in the non-recording case.

There are 2 ways to go about this:

* Just format anyway.
* Introduce some sort of "lazy attributes" that's only evaluated by the span processor. `{'$lazy', fun()}`?

**Comment by tsloughter** [comment]
I think just format (`lists:concat(?FUNCTION_NAME, "/", ?FUNCTION_ARITY)`) it anyway and we should look at proposing `?FUNCTION_NAME_STRING` that includes arity to OTP.
